### PR TITLE
Add the block fetch mock server to the peer simulator

### PR DESCRIPTION
**Pull request by facundominguez**

### CODE REVIEW

File: ouroboros-consensus/test/infra-test/Test/Ouroboros/Consensus/ChainGenerator/Tests/Sync.hs [hunk 1]
```diff
@@ -425,23 +463,7 @@ mkChainDb tracer candidateVars nodeCfg registry = do
         (\_ -> ChainDB.Impl.openDBInternal chainDbArgs False)
         (ChainDB.closeDB . fst)
     _ <- forkLinkedThread registry "AddBlockRunner" intAddBlockRunner
-    void $ flip Map.traverseWithKey candidateVars $ \ pid varCandidate ->
-      forkLinkedThread registry (condense pid) $ monitorCandidate chainDB varCandidate
     pure chainDB
-  where
-    monitorCandidate chainDB varCandidate =
```
Comment (tek) [inline ouroboros-consensus/test/infra-test/Test/Ouroboros/Consensus/ChainGenerator/Tests/Sync.hs:432]: This was the simplified mock of what is now done by `Ouroboros.Network.BlockFetch.Client.blockFetchClient`?

File: ouroboros-consensus/test/infra-test/Test/Ouroboros/Consensus/PeerSimulator/BlockFetch.hs [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+  -> FetchClientRegistry peer (Header TestBlock) TestBlock m
+  -> STM m (Map peer (AnchoredFragment (Header TestBlock)))
+  -> m ()
+startBlockFetchLogic registry chainDb fetchClientRegistry getCandidates = do
+    let slotForgeTime :: BlockFetchClientInterface.SlotForgeTimeOracle m TestBlock
+        slotForgeTime _ = pure dawnOfTime
+
+        blockFetchConsensusInterface =
+          BlockFetchClientInterface.mkBlockFetchConsensusInterface
+            (TestBlockConfig $ NumCoreNodes 10) -- TODO: what's this
```
Comment (tek) [inline ouroboros-consensus/test/infra-test/Test/Ouroboros/Consensus/PeerSimulator/BlockFetch.hs:67]: didn't @amesgen say that this is only relevant for minting nodes?

**Comment by tek** [comment]
Looks reasonable enough to me, but I'm not exactly an authority on how BlockFetch should look. @amesgen did you sign off on the entire implementation in the sync call?

**Review by tek — APPROVED** [review]

File: ouroboros-consensus/test/infra-test/Test/Ouroboros/Consensus/ChainGenerator/Tests/Sync.hs (hunk 1 above)
Comment (facundominguez) [reply ouroboros-consensus/test/infra-test/Test/Ouroboros/Consensus/ChainGenerator/Tests/Sync.hs:432]: Yes, this was the exchanging of blocks without block fetch.

File: ouroboros-consensus/test/infra-test/Test/Ouroboros/Consensus/PeerSimulator/BlockFetch.hs (hunk 2 above)
Comment (facundominguez) [reply ouroboros-consensus/test/infra-test/Test/Ouroboros/Consensus/PeerSimulator/BlockFetch.hs:67]: Yes, used for some round robin when deciding the leader schedule. I'll remove the comment.

**Comment by facundominguez** [comment]
I rebased after merging #371.

**Pull request merged**